AMDirT/core/ena.py

```diff
@@ -0,0 +1,206 @@
+        return json_resp
+
+    def list_fields(self, result_type: str) -> None:
+        """Display list of available fields
+
+        Args:
+            result_type(str): A result is a set of data that can
+            be searched against and returned
+        Returns:
+            List: list of available fields
+        """
+        json_resp = self._get_fields(result_type=result_type)
+        for field in json_resp:
+            logger.info(f"{field['columnId']} - {field['description']}")
+        logger.info(f"Available fields for {result_type} are: {json_resp}")
+
+    def _check_result_type(self, result_type: str) -> bool:
+        """Check if result type is allowed
+
+        Args:
+            result_type(str): A result is a set of data that can
+            be searched against and returned
+        Returns:
+            bool: True if result type is valid, False otherwise
+        """
+        all_results = self._get_results()
+        results = [result["resultId"] for result in all_results]
+        if result_type not in results:
+            logger.warning(f"{result_type} is not a valid result type")
+            return False
+        return True
+
+    def _check_fields(self, result_type: str, fields: List[str]) -> bool:
+        """Check if fields are allowed
+
+        Args:
+            result_type(str): A result is a set of data that can
+            be searched against and returned
+            fields(List): list of fields to check
+        Returns:
+            bool: True if fields are valid, False otherwise
+        """
+        all_fields = self._get_fields(result_type)
+        fields = [field["columnId"] for field in all_fields]
+        for field in fields:
+            if field not in fields:
+                logger.warning(f"{field} is not a valid field")
+                return False
+        return True
+
+    def query(
+        self,
+        accession: str,
+        result_type: str = "read_run",
+        fields: List = [
+            "run_accession",
+            "sample_accession",
+            "fastq_ftp",
+            "fastq_md5",
+            "fastq_bytes",
+        ],
+    ) -> dict:
+        """Generate list of runs metadata for a study accession
+
+        Args:
+            accession (str): ENA accession
+            result_type(str): A result is a set of data that can
+            be searched against and returned
+            fields(List): list of fields to return
+
+        Returns:
+            dict: run_accession as keys, and metadata as values
+        """
+
+        self._check_result_type(result_type)
+        self._check_fields(result_type, fields)
+        url = os.path.join(
+            self.base_url,
+            f"filereport?accession={accession}&download=false&format=json&result={result_type}&fields={','.join(fields)}",
+        )
+        json_resp = self.__get_json__(url)
+        return json_resp
+
+
+if __name__ == "__main__":
+    e = ENAPortalAPI()
+    print(e.query("PRJEB30331"))
```

Comment:
Is this just an example, or is it meant to be hardcoding?

Reply:
This is just an example :)